ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionIndexing.scala

```diff
@@ -250,31 +226,33 @@ object TransactionIndexing {
 
   final case class EventsInfo(
       events: Vector[(NodeId, Node)],
-      archives: Set[ContractId],
       stakeholders: WitnessRelation[NodeId],
       disclosure: WitnessRelation[NodeId],
+      divulgedContractInfos: Iterable[DivulgedContractInfo],
   )
 
-  final case class ContractsInfo(
-      netCreates: Set[Create],
-      netArchives: Set[ContractId],
-      divulgedContracts: Iterable[DivulgedContract],
-  )
-
-  final case class ContractWitnessesInfo(
-      netArchives: Set[ContractId],
-      netVisibility: WitnessRelation[ContractId],
+  // most involved change: divulgence is now an event instead encoded in contract/contract_witness tables
+  // also we generate these events directly as part of the other events, so we need to land the necessery
+  // data in the EventsTablePostgresql
+  case class DivulgedContractInfo(
+      contractId: ContractId,
+      contractInst: Option[
+        ContractInst
+      ], // this we do not necessarily have: for public KV ledgers the divulged contract list is empty, since all contracts must be at all ledger as create nodes already
```

Comment:
```suggestion
      ], // this we do not necessarily have: for public KV ledgers and sanbox-classic the divulged contract list is empty, since all contracts must be at all ledger as create nodes already
```